packages/ra-ui-materialui/src/input/ArrayInput/SimpleFormIterator.tsx

```diff
@@ -166,19 +190,39 @@ export const SimpleFormIterator = (props: SimpleFormIteratorProps) => {
                         </SimpleFormIteratorItem>
                     ))}
                 </ul>
-                {!disabled && !disableAdd && (
-                    <div className={SimpleFormIteratorClasses.add}>
-                        {cloneElement(addButton, {
-                            className: clsx(
-                                'button-add',
-                                `button-add-${source}`
-                            ),
-                            onClick: handleAddButtonClick(
-                                addButton.props.onClick
-                            ),
-                        })}
-                    </div>
-                )}
+                <div className={SimpleFormIteratorClasses.line}>
+                    {!disabled && !disableAdd && (
+                        <div className={SimpleFormIteratorClasses.add}>
+                            {cloneElement(addButton, {
+                                className: clsx(
+                                    'button-add',
+                                    `button-add-${source}`
+                                ),
+                                onClick: handleAddButtonClick(
+                                    addButton.props.onClick
+                                ),
+                            })}
+                        </div>
+                    )}
+                    {fields.length > 0 && (
+                        <div className={SimpleFormIteratorClasses.clear}>
+                            {cloneElement(clearButton, {
```

Comment:
I know the add button uses `cloneElement`, but that's a leftover from react-admin 3.X. We try not to clone elements anymore in 4.0.

Reply:
All buttons(`remove`, `add`, `reorder`) for `SimpleFormIterator` are using `cloneElement`, so, to use `cloneElement` is for new feature(`clear`) is more attractive for me. To be in sameness.

I understand your concern, however, these may be able to be refactored in the future, but it does not related to the purpose of this PR, I guess.

Reply:
No, because using cloneElement today is adding debt, and preventing us from changing it in a backwards compatible way. New solutions should not use cloneElement from day one.

Reply:
resolve commit : 3669f3a